Lib/ldap/constants.py

```diff
@@ -0,0 +1,402 @@
+    TLSInt('OPT_X_TLS_CERTFILE'),
+    TLSInt('OPT_X_TLS_KEYFILE'),
+    TLSInt('OPT_X_TLS_REQUIRE_CERT'),
+    TLSInt('OPT_X_TLS_CIPHER_SUITE'),
+    TLSInt('OPT_X_TLS_RANDOM_FILE'),
+    TLSInt('OPT_X_TLS_DHFILE'),
+    TLSInt('OPT_X_TLS_NEVER'),
+    TLSInt('OPT_X_TLS_HARD'),
+    TLSInt('OPT_X_TLS_DEMAND'),
+    TLSInt('OPT_X_TLS_ALLOW'),
+    TLSInt('OPT_X_TLS_TRY'),
+    TLSInt('OPT_X_TLS_PEERCERT', optional=True),
+
+    TLSInt('OPT_X_TLS_VERSION', optional=True),
+    TLSInt('OPT_X_TLS_CIPHER', optional=True),
+    TLSInt('OPT_X_TLS_PEERCERT', optional=True),
+
+    # only available if OpenSSL supports it => might cause
+    # backward compability problems
+    TLSInt('OPT_X_TLS_CRLCHECK', optional=True),
+
+    TLSInt('OPT_X_TLS_CRLFILE', optional=True),
+
+    TLSInt('OPT_X_TLS_CRL_NONE'),
+    TLSInt('OPT_X_TLS_CRL_PEER'),
+    TLSInt('OPT_X_TLS_CRL_ALL'),
+    TLSInt('OPT_X_TLS_NEWCTX', optional=True),
+    TLSInt('OPT_X_TLS_PROTOCOL_MIN', optional=True),
+    TLSInt('OPT_X_TLS_PACKAGE', optional=True),
+
+    Int('OPT_X_SASL_MECH'),
+    Int('OPT_X_SASL_REALM'),
+    Int('OPT_X_SASL_AUTHCID'),
+    Int('OPT_X_SASL_AUTHZID'),
+    Int('OPT_X_SASL_SSF'),
+    Int('OPT_X_SASL_SSF_EXTERNAL'),
+    Int('OPT_X_SASL_SECPROPS'),
+    Int('OPT_X_SASL_SSF_MIN'),
+    Int('OPT_X_SASL_SSF_MAX'),
+    Int('OPT_X_SASL_NOCANON', optional=True),
+    Int('OPT_X_SASL_USERNAME', optional=True),
+    Int('OPT_CONNECT_ASYNC', optional=True),
+    Int('OPT_X_KEEPALIVE_IDLE', optional=True),
+    Int('OPT_X_KEEPALIVE_PROBES', optional=True),
+    Int('OPT_X_KEEPALIVE_INTERVAL', optional=True),
+
+    Int('DN_FORMAT_LDAP'),
+    Int('DN_FORMAT_LDAPV3'),
+    Int('DN_FORMAT_LDAPV2'),
+    Int('DN_FORMAT_DCE'),
+    Int('DN_FORMAT_UFN'),
+    Int('DN_FORMAT_AD_CANONICAL'),
+    # Int('DN_FORMAT_LBER'),  # for testing only
+    Int('DN_FORMAT_MASK'),
+    Int('DN_PRETTY'),
+    Int('DN_SKIP'),
+    Int('DN_P_NOLEADTRAILSPACES'),
+    Int('DN_P_NOSPACEAFTERRDN'),
+    Int('DN_PEDANTIC'),
+
+    Int('AVA_NULL'),
+    Int('AVA_STRING'),
+    Int('AVA_BINARY'),
+    Int('AVA_NONPRINTABLE'),
+
+    Int('OPT_SUCCESS'),
+
+    # XXX - these should be errors
+    Int('URL_ERR_BADSCOPE'),
+    Int('URL_ERR_MEM'),
+    # Int('LIBLDAP_R'),
+
+    Feature('LIBLDAP_R', 'HAVE_LIBLDAP_R'),
+    Feature('SASL_AVAIL', 'HAVE_SASL'),
+    Feature('TLS_AVAIL', 'HAVE_TLS'),
+
+    Str("CONTROL_MANAGEDSAIT"),
+    Str("CONTROL_PROXY_AUTHZ"),
+    Str("CONTROL_SUBENTRIES"),
+    Str("CONTROL_VALUESRETURNFILTER"),
+    Str("CONTROL_ASSERT"),
+    Str("CONTROL_PRE_READ"),
+    Str("CONTROL_POST_READ"),
+    Str("CONTROL_SORTREQUEST"),
+    Str("CONTROL_SORTRESPONSE"),
+    Str("CONTROL_PAGEDRESULTS"),
+    Str("CONTROL_SYNC"),
+    Str("CONTROL_SYNC_STATE"),
+    Str("CONTROL_SYNC_DONE"),
+    Str("SYNC_INFO"),
+    Str("CONTROL_PASSWORDPOLICYREQUEST"),
+    Str("CONTROL_PASSWORDPOLICYRESPONSE"),
+    Str("CONTROL_RELAX"),
+)
+
+
+def write_header():
```

Comment:
It does a bit more than just writing the header :)

Reply:
I don't understand the comment. It dumps to stdout in C header format, nothing more :)
I've renamed it to `print_header` to point out the output is not configurable.

Reply:
Ah, you refer to C header. I though that you were referring to file header `` Generated with: ...``. Forget what I said. :)